src/main/kotlin/org/jetbrains/plugins/bsp/ui/gutters/BspJVMRunLineMarkerContributor.kt

```diff
@@ -30,8 +37,12 @@ public class BspJVMRunLineMarkerContributor : RunLineMarkerContributor() {
     else null
 
   private fun PsiElement.shouldAddMarker(): Boolean =
-    !isInsideJar() && getStrictParentOfType<PsiNameIdentifierOwner>()
-      ?.isClassOrMethod() ?: false
+    (!isInsideJar() && getStrictParentOfType<PsiNameIdentifierOwner>()
+      ?.isClassOrMethod() ?: false) || isGoTopLevelFunction()
+
+  private fun PsiElement.isGoTopLevelFunction() =
```

Comment:
i believe that u need to create a separate class for go (to register it correctly in plugin.xml) but prob it'd be worth to create a base class and just 2 impls: jvm and go